tests/checkers/rst_style.py

```diff
@@ -137,12 +139,13 @@ def get_next_line_first_word(next_line_):
     for lno, line in enumerate(lines):
         if lno + 1 < len(lines):
             next_line = lines[lno + 1]
-            if (is_valid_line(line, ('+', '|'))
+            if (
+                is_valid_line(line, ('+', '|'))
                 and is_valid_line(next_line, ('+', '|', '- ', '* ', '#. '))
             ):
-                current_line_remaining_space = options.max_line_length - len(line)
+                current_line_remaining_space = options.max_line_length - len(line.rstrip())
                 next_line_first_word = get_next_line_first_word(next_line).rstrip()
-                if current_line_remaining_space > len(next_line_first_word):
+                if current_line_remaining_space >= len(next_line_first_word):
```

Comment:
This part doesn't seem to be about separators :thinking:
What are you trying to fix here?

Reply:
It's a separate fix; I'll split the commit.

`current_line_remaining_space` was wrong as it was counting the `\n` character too.

However, the second change is probably wrong as we need to accommodate space for... the space char too.